demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -41,26 +58,36 @@
  * This demo uses the Mosquitto test server. This is a public MQTT server; do not
  * publish anything sensitive to this server.
  */
-#define SERVER                 "test.mosquitto.org"
+#define BROKER_ENDPOINT           "test.mosquitto.org"
+
+/**
+ * @brief Length of MQTT server host name.
+ */
+#define BROKER_ENDPOINT_LENGTH    ( ( uint16_t ) ( sizeof( BROKER_ENDPOINT ) - 1 ) )
 
 /**
  * @brief MQTT server port number.
  *
  * In general, port 8883 is for secured MQTT connections.
  */
-#define PORT                   8883
+#define BROKER_PORT               8883
 
 /**
  * @brief Path of the file containing the server's root CA certificate.
  *
  * This certificate should be PEM-encoded.
  */
-#define SERVER_CERT            "mosquitto.org.crt"
+#define SERVER_CERT_PATH          "mosquitto.org.crt"
```

Comment:
Is it okay if we set `SERVER_CERT_PATH` to `certificates/mosquitto.org.crt`? Because the certificate needs to have the same path relative to the executable when compiling using `cmake`. Otherwise, we have to put `mosquitto.org.crt` together with other executables in the same directory.

Reply:
Yea, I thought about that. Yea, that is a good idea.

Reply:
Cool. If you add this to the end of the `CMakeLists.txt` file for this demo, it will automatically be copied over:
```cmake
add_custom_command(
        TARGET ${DEMO_NAME} POST_BUILD
        COMMAND ${CMAKE_COMMAND} -E copy
                ${CMAKE_CURRENT_SOURCE_DIR}/certificates/mosquitto.org.crt
                ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/certificates/mosquitto.org.crt)
```

Reply:
I was also thinking the same. Just did that in the latest patch.